addons/sale_management/models/res_config_settings.py

```diff
@@ -8,11 +8,35 @@ class ResConfigSettings(models.TransientModel):
     _inherit = 'res.config.settings'
 
     group_sale_order_template = fields.Boolean("Quotation Templates", implied_group='sale_management.group_sale_order_template')
-    default_sale_order_template_id = fields.Many2one('sale.order.template', default_model='sale.order', string='Default Template')
+    company_so_template_id = fields.Many2one(
+        related="company_id.sale_order_template_id", string="Default Template", readonly=False,
+        domain="['|', ('company_id', '=', False), ('company_id', '=', company_id)]")
     module_sale_quotation_builder = fields.Boolean("Quotation Builder")
 
     @api.onchange('group_sale_order_template')
     def _onchange_group_sale_order_template(self):
         if not self.group_sale_order_template:
             self.module_sale_quotation_builder = False
-            self.default_sale_order_template_id = False
+
+    @api.model
+    def create(self, values):
```

Comment:
seems louche, should perhaps be a override of `default_get` and `execute` instead